### PR TITLE
fix: MsgDeposit trans interpolation [web-cosmos]

### PR DESCRIPTION
**Pull request by rachelhox**
## Description

Closes: [BDU-778](https://forbole.atlassian.net/browse/BDU-778)

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] ran linting via `yarn lint`
- [x] wrote tests where necessary
- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] targeted the correct branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [x] added a changeset via [`yarn && yarn changeset`](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)

[BDU-778]: https://forbole.atlassian.net/browse/BDU-778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/lovely-foxes-sell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'ui': minor
+---
+
+fix MsgDeposit type interpolation
```
Comment (teamchong) [inline .changeset/lovely-foxes-sell.md:5]: It appears that the this changeset is identical to the "fix MsgDeposit interpolation issue" changeset. 
Can you clarify the intention or purpose of this?

File: packages/ui/src/components/msg/governance/deposit_proposal/index.tsx [hunk 2]
```diff
@@ -5,8 +5,12 @@ import { formatNumber, formatToken } from '@/utils/format_token';
 import { PROPOSAL_DETAILS } from '@/utils/go_to_page';
 import Typography from '@mui/material/Typography';
 import { Trans, useTranslation } from 'next-i18next';
-import Link from 'next/link';
-import { FC, useMemo } from 'react';
+import Link, { LinkProps } from 'next/link';
+import { FC } from 'react';
+
+const LinkText = ({ href, children }: React.PropsWithChildren<LinkProps>) => (
```
Comment (teamchong) [inline packages/ui/src/components/msg/governance/deposit_proposal/index.tsx:11]: Which style do you prefer: the current one or the following?
```
import { FC, PropsWithChildren } from 'react';
... : PropsWithChildren<LinkProps>
```
I like the shorter code, but I know most people use the `React.` prefix

File: .changeset/lovely-foxes-sell.md (hunk 1 above)
Comment (rachelhox) [reply .changeset/lovely-foxes-sell.md:5]: yeah it's a duplicate, will remove the redundant changeset, thanks!

File: packages/ui/src/components/msg/governance/deposit_proposal/index.tsx (hunk 2 above)
Comment (rachelhox) [reply packages/ui/src/components/msg/governance/deposit_proposal/index.tsx]: ok modified the style [e7defdb](https://github.com/forbole/big-dipper-2.0-cosmos/pull/1175/commits/e7defdbc2e1479ec48261f78ba3f61c7952e5a1f)

**Review by teamchong — DISMISSED** [review]
image
image
https://cosmos-git-ui-msgdeposit-bigdipper.vercel.app/transactions/C1DA70E7319F8BFE66A714B8AB29DB9E3472F264EED9CCF7F6E331EF7D1CDC8B

LGTM

File: packages/ui/src/components/msg/governance/deposit_proposal/index.tsx [hunk 3]
```diff
@@ -5,9 +5,17 @@ import { formatNumber, formatToken } from '@/utils/format_token';
 import { PROPOSAL_DETAILS } from '@/utils/go_to_page';
 import Typography from '@mui/material/Typography';
 import { Trans, useTranslation } from 'next-i18next';
-import Link from 'next/link';
-import { FC, useMemo } from 'react';
+import Link, { LinkProps } from 'next/link';
+import { FC, ReactNode } from 'react';
```
Comment (teamchong) [inline packages/ui/src/components/msg/governance/deposit_proposal/index.tsx:9]:
```suggestion
import { FC, PropsWithChildren, ReactNode } from 'react';
```

File: packages/ui/src/components/msg/governance/deposit_proposal/index.tsx [hunk 4]
```diff
@@ -5,9 +5,17 @@ import { formatNumber, formatToken } from '@/utils/format_token';
...
 
+const LinkText: FC<{ href: LinkProps['href']; children: ReactNode }> = (props) => {
```
Comment (teamchong) [inline packages/ui/src/components/msg/governance/deposit_proposal/index.tsx:11]:
```suggestion
const LinkText: FC<PropsWithChildren<{ href: LinkProps['href'] >> = (props) => {
```

**Review by teamchong — COMMENTED** [review]
an alternative way to add typing for props with children

File: packages/ui/src/components/msg/governance/deposit_proposal/index.tsx (hunk 4 above)
Comment (rachelhox) [reply packages/ui/src/components/msg/governance/deposit_proposal/index.tsx:11]: which one do you prefer for style? `children: ReactNode` or `PropsWithChildren`?

Comment (teamchong) [reply packages/ui/src/components/msg/governance/deposit_proposal/index.tsx:11]: I leave it up to you to decide, `PropsWithChildren` seems to be the more popular choice https://blog.logrocket.com/using-react-children-prop-with-typescript/

**Review by teamchong — APPROVED** [review]
GLTM

**Pull request merged**